demisto_sdk/dev_tools/spell_checker.py

```diff
@@ -1,95 +1,129 @@
-            return 1
+        if self.unknown_words:
+            print_error(u"Words that might be misspelled were found in {}:\n\n{}".format(
+                self.path, '\n'.join(self.unknown_words)))
+            return False
 
-        print_color("No problematic words found in {}".format(self.path), LOG_COLORS.GREEN)
-        return 0
+        print_color("No misspelled words found in {}".format(self.path), LOG_COLORS.GREEN)
+        return True
 
-    def check_md_file(self, md_data, unknown_words):
+    def check_md_file(self, md_file_lines):
         """Runs spell check on .md file. Adds unknown words to given unknown_words set.
 
         Args:
-            md_data (list): A line list of the .md file contents
-            unknown_words (set): A set of found unknown words
-
-        Returns:
-            None.
+            md_file_lines (list): A line list of the .md file contents
         """
-        for line in md_data:
+        for line in md_file_lines:
             for word in line.split():
                 if word.isalpha() and self.spellchecker.unknown([word]):
-                    unknown_words.add(word)
+                    self.unknown_words.add(word)
 
-    def check_yaml(self, yml_info, unknown_words):
+    def check_yaml(self, yml_info):
         """Runs spell check on .yml file. Adds unknown words to given unknown_words set.
 
         Args:
             yml_info (Loader): A line list of the .md file contents
-            unknown_words (set): A set of found unknown words
-
-        Returns:
-            None.
         """
+        # yml file is parsed as a large dictionary. Each key in the large dictionary
+        # has a value which can be comprised of a string, a sub-dictionary or a list of dictionaries.
+        # The following code separates the spell check to these cases.
         for key, value in yml_info.items():
-            if key in DISPLAYABLE_LINES:
+            # case 1: a String.
+            if key in USER_VISIBLE_LINE_KEYS:
+
+                # separate the string to individual words
                 for word in value.split():
+
+                    # Drop commas, full-stops and brackets.
+                    if word.endswith(',') or word.endswith('.') or word.endswith(')'):
+                        word = word[:-1]
+
+                    if word.startswith('('):
+                        word = word[1:]
+
+                    # if a word is comprised of only letters (no punctuation ia checked!)
                     if word.isalpha() and self.spellchecker.unknown([word]):
-                        unknown_words.add(word)
+                        self.unknown_words.add(word)
 
             else:
+                # case 2: a sub-dictionary
                 if isinstance(value, dict):
-                    if key != SCRIPT_ARGS:
-                        self.check_yaml(self.spellchecker, value, unknown_words)
+                    # ignore command arguments in playbooks - no need to check these.
```

Comment:
Let's add reference to "scriptarguments" here

Reply:
it's kind of long - I dont think we want to do it
scriptarguments:
      message-id:
        complex:
          root: Gmail
          filters:
          - - operator: isNotEmpty
              left:
                value:
                  simple: Gmail.Attachments
                iscontext: true
          transformers:
          - operator: GetFirstObject
          - operator: getField
            args:
              field:
                value:
                  simple: ID

Reply:
I meant we should add a small reference to what "scriptarguments" is

Reply:
done